src/teleop/gui/src/components/AutonTask.vue

```diff
@@ -1,20 +1,315 @@
 <template>
-<div>
-    <p>Auton</p>
+<div class="wrapper">
+    <div class="box header">
+      <img src="/static/mrover.png" alt="MRover" title="MRover" width="48" height="48" />
+      <h1>Auton Dashboard</h1>
+      <div class="spacer"></div>
+      <div class="spacer"></div>
+      <div class="help">
+        <img src="/static/help.png" alt="Help" title="Help" width="48" height="48" />
+      </div>
+      <div class="helpscreen"></div>
+      <div class="helpimages" style="display: flex; align-items: center; justify-content: space-evenly">
+        <img src="/static/joystick.png" alt="Joystick" title="Joystick Controls" style="width: auto; height: 70%; display: inline-block" />
+      </div>
+    </div>
+    <div class="box1 data" v-bind:style="{backgroundColor: nav_state_color}">
+      <h2>Nav State: {{this.nav_status.nav_state_name}}</h2>
+    </div>
+    <div class="box map light-bg">  
+      <AutonRoverMap v-bind:odom="odom" v-bind:TargetBearing="TargetBearing"/>
+    </div>
+    <div class="box waypoints light-bg">
+      <AutonWaypointEditor v-bind:odom="odom" v-bind:AutonDriveControl="AutonDriveControl"/>
+    </div>
+    <!--Enable the drive controls if auton is off-->
+    <div class="driveControls" v-if="!this.autonEnabled" v-show="false">
+      <DriveControls/>
+    </div>
 </div>
 </template>
 
 <script>
+import ROSLIB from "roslib"
+import AutonRoverMap from "./AutonRoverMap.vue"
+import AutonWaypointEditor from './AutonWaypointEditor.vue'
+import DriveControls from "./DriveControls.vue";
+import { mapGetters } from 'vuex';
+import * as qte from "quaternion-to-euler";
+
+const navBlue = "#4695FF"
+const navGreen = "yellowgreen"
+const navRed = "lightcoral"
+const navGrey = "lightgrey"
 
 export default {
   data() {
     return {
 
+      // Default coordinates are at NC 53 Parking Lot
+      odom: {
+        latitude_deg: 42.294864932393835,
+        longitude_deg: -83.70781314674628,
+        bearing_deg: 0
+      },
+
+      TargetBearing: {
+        target_bearing: 0
+      },
+
+      // Current Values being output to drivetrain for auton
+      AutonDriveControl: {
+        left_percent_velocity: 0,
+        right_percent_velocity: 0
+      },
+
+      nav_status: {
+        nav_state_name: "Off",
+        completed_wps: 0,
+        total_wps: 0
+      },
+
+      enableAuton: {
+        enable: false,
+        GPSWaypoint: []
+      },
+
+      navBlink: false,
+      greenHook: false,
+
+      // Pubs and Subs
+      nav_status_sub: null,
```

Comment:
I think we just gotta choose a casing standard (snake vs camel) for things like member variables, non-member variables, and functions, and just be consistent

Reply:
Yeah eslint would have us covered in that respect but it's kind of blocked rn.

Reply:
Yeah iirc theres a setting for it to enforce camel case across the board